agent/resources_monitor.go

```diff
@@ -0,0 +1,135 @@
+package agent
+
+import (
+	"context"
+	"time"
+
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+	"github.com/coder/coder/v2/agent/proto"
+	"github.com/coder/coder/v2/cli/clistat"
+	"github.com/coder/quartz"
+)
+
+func (a *agent) pushResourcesMonitoring(ctx context.Context, aAPI proto.DRPCAgentClient24) error {
```

Comment:
self review : We have this intermediate function in order to have an extra layer, and simplicify for testing purpose. 
As all arguments are injected as parameters to `PushResourcesMonitoringWithConfig` we can use the mocked versions.